lib/index.ts

```diff
@@ -184,38 +184,25 @@ export default class Backburner {
 
   public run(method: Function);
   public run(target: Function | any | null, method?: Function | string, ...args);
-  public run(target: any | null | undefined, method?: any, ...args: any[]) {
-    let length = arguments.length;
-    let _method: Function | string;
-    let _target: any | null | undefined;
-
-    if (length === 1) {
-      _method = <Function> target;
-      _target = null;
-    } else {
-      _method = method;
-      _target = target;
-
-      if (isString(_method)) {
-        _method = <Function> _target[_method];
-      }
-    }
+  public run(target: any | null | undefined, method?: any, ...args: any[]);
+  public run() {
+    let [target, method, args] = this.parseArgs(arguments);
```

Comment:
In general, we want to avoid passing arguments out to other methods like this.  I also don't think `this` is needed in `parseArgs`, so you could refactor to use a utility method and avoid the `arguments` deopt.

This would be something like:

```js
function parseArgs() {
  // existing logic
}

// ...snip...
public run() {
  let [target, method, args] = parseArgs.apply(null, arguments);
  // ...snip...
}
```